varcode/variant.py

```diff
@@ -223,9 +223,13 @@ def __setstate__(self, fields):
         self.ensembl = cached_release(fields.pop("release"))
         self.info = fields.pop("info", {})
 
-        # Remaining fields are simple properties that just get set.
+        # Remaining fields  are simple properties that just get set.
         self.__dict__.update(fields)
 
+        # TODO: switch to interbase coordinates so that 'end' becomes a
+        # mandatory rather than derived field
+        self.end = self.start + max(0, len(self.ref) - 1)
```

Comment:
This line may be right but it gives me pause. Can we add `end` to `fields()` instead? Then we would just extract it from the saved state here.